### ISSUE TITLE
Remove moment.js

### ISSUE THREAD

**Issue by maggie44** [balena-io/balena-sdk#1219]
Moment.js is largely deprecated. It's own README included in the installed package states:

```
## Project Status

Moment.js is a legacy project, now in maintenance mode.  In most cases, you should choose a different library.

For more details and recommendations, please see [Project Status](https://momentjs.com/docs/#/-project-status/) in the docs.
```

It looks like in the SDK it is only used here: https://github.com/balena-io/balena-sdk/blob/master/lib/util/date.ts

It would be nice to switch to something else, because the Balena SDK is already the largest node_module in my projects, followed only by `moment`:

![Screenshot 2022-04-23 at 20 28 25](https://user-images.githubusercontent.com/64841595/164943487-410892bd-c0ad-42f2-a2f7-1a777e7eb434.png)

When size matters, catching these low hanging fruit makes a difference. 

---
```
yarn why moment
└─ balena-sdk@npm:16.19.7
   └─ moment@npm:2.29.2 (via npm:^2.29.1)
```

**Comment by thgreasi** [comment]
Thanks for raising this. I will pass this internally and prioritize accordingly.

**Comment by thgreasi** [comment]
Resolved by #1265 in v16.26.4
